packages/@webex/plugin-cc/test/unit/spec/cc.ts

```diff
@@ -1,30 +1,286 @@
-import ContactCenter from '@webex/plugin-cc/src';
-import Mercury from '@webex/internal-plugin-mercury';
+import { assert } from '@webex/test-helper-chai';
 import MockWebex from '@webex/test-helper-mock-webex';
-import {Services} from '@webex/webex-core';
+import WebSocket from '../../../src/WebSocket/WebSocket';
+import { EVENT, READY, WCC_API_GATEWAY, WEBSOCKET_EVENT_TIMEOUT } from '../../../src/constants';
+import { CC_EVENTS } from '../../../src/types';
+import ContactCenter from '../../../src/cc';
 
-describe('CC Tests', () => {
+describe('webex.cc', () => {
   let webex;
+  let webSocketMock;
+  let eventEmitter;
 
   beforeEach(() => {
     webex = new MockWebex({
       children: {
         cc: ContactCenter,
-        mercury: Mercury,
-        services: Services,
       },
     });
-  })
 
-  it('Success: invoke the register function, resolved promise', () => {
-    webex.cc.register(true).then((data) => {
-        expect(data).toEqual('Success: Dummy data returned');
-    })
+    // Ensure webex.internal.services is initialized correctly
+    webex.internal = {
+      ...webex.internal,
+      services: {
+        get: jest.fn().mockReturnValue('https://api.example.com/'),
+      },
+    };
+
+    // Manually mock the necessary methods
+    webSocketMock = {
+      on: jest.fn(),
+      subscribeAndConnect: jest.fn(),
+      disconnectWebSocket: jest.fn(),
+      off: jest.fn(),
+    };
+    webex.cc.webSocket = webSocketMock;
+    
+    // Mock event emitter
+    eventEmitter = new (require('events')).EventEmitter();
+    webSocketMock.on.mockImplementation((event, callback) => {
+      eventEmitter.on(event, callback);
+    });
+
+    global.WEBSOCKET_EVENT_TIMEOUT = 100; // Set to a smaller value for tests
+
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  describe('#register', () => {
+
+    it('should resolve with success message on successful registration even if $config is undefined', async () => {
+      webex.$config = undefined;
+      const promise = webex.cc.register();
+
+      // Emit the welcome event to simulate the WebSocket message
+      eventEmitter.emit(EVENT, {
+        type: CC_EVENTS.WELCOME,
+        data: { agentId: 'mockAgentId' },
+      });
+
+      const result = await promise;
+      expect(webSocketMock.subscribeAndConnect).toHaveBeenCalled();
```

Comment:
These 2 expect statements can be combined into one and use toHaveBeenCalledOnceWith. Applies everywhere